### ISSUE TITLE
BACnet over IP

### ISSUE THREAD

**Issue by andreapx** [JoelBender/bacpypes-mqtt#4]
Hi!
Is there a way to use this gateway with BACnet over IP?
Thanks!

**Comment by JoelBender** [comment]
Yes, checkout the [mqtt2ip](https://github.com/JoelBender/bacpypes-mqtt/blob/master/mqtt2ip.pdf) sample application which acts as a router between a broker and a BACnet/IPv4 network.  This is conceptually the same as an IPv4-to-SC router, so if you have two of these applications running on two different BACnet networks (like two different IPv4 subnets) you can route traffic between them going through the broker.  By running the console_client application you will be a BACnet device on the virtual network inside the broker and you can talk to any of the other MQTT clients or through the routers to BACnet/IPv4 devices.

This is designed to be able to use any broker you would like, like a commercial cloud service that uses TLS and authentication.

**Comment by andreapx** [comment]
Thanks!!! I'll try that!

**Comment by andreapx** [comment]
Hi Joel,
I'm trying to use your software, but I'm having some troubles (I'm not an expert in python/pip/etc.), let's see if you can give me a hand!
If I run `pipenv run python console_client.py` and I run the command `whois`, in MQTT I just get some garbage:

![07-07-2023_11-03-36_6_MQTT_Explorer](https://github.com/JoelBender/bacpypes-mqtt/assets/13846728/60f57c31-1b7e-4710-bfca-451b2943e35b)

and no response on the console.
I've tried running YABE (Yet Another Bacnet Explorer) I can see the BacNet device (server) and all the data.
What am I doing wrong?
Thanks!
